added login

DhruviJ17 opened the pull request:
### Description

<!--Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change.-->

Added link of contributing guidelines

Fixes #47 


### Type of Change:
<!--**Delete irrelevant options.**-->

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)



### How Has This Been Tested?
<!--Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.-->


### Checklist:
<!--**Delete irrelevant options.**-->

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable


**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

devkapilbansal reviewed (commented):
@DhruviJ17 which issue you are working on. Please link that issue too in the pull request and enforce issue closing in commit message. If there is no such issue, consider opening one and wait till the issue is accepted and assigned to you.

DhruviJ17 commented:
I am just trying to add new features which I thought were missing!

On Mon, Nov 9, 2020, 1:37 PM Kapil Bansal <notifications@github.com> wrote:

> *@devkapilbansal* commented on this pull request.
>
> @DhruviJ17 <https://github.com/DhruviJ17> which issue you are working on.
> Please link that issue too in the pull request and enforce issue closing in
> commit message. If there is no such issue, consider opening one and wait
> till the issue is accepted and assigned to you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/anitab-org/open-source-programs-backend/pull/43#pullrequestreview-526005204>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQOPSVYTTVF2INJONPOY533SO6PL3ANCNFSM4TNO7Y5A>
> .
>

devkapilbansal reviewed (changes requested):
Thanks @DhruviJ17 
Please open an issue for this too. Also, edit the pull request message template and fill relevant information there

DhruviJ17 commented:
@devkapilbansal I have opened an issue for the above so please can you accept it and assign it to me

devkapilbansal commented:
> @devkapilbansal I have opened an issue for the above so please can you accept it and assign it to me

@DhruviJ17 I can't accept it. Consider asking it on zulip

isabelcosta commented:
@DhruviJ17 can you please post the link in a comment here, or edit the PR (pull request) description to include that?

DhruviJ17 commented:
@isabelcosta  this is the link for the issue created by me: https://github.com/anitab-org/open-source-programs-backend/issues/47

isabelcosta commented on token_auth/views/password_reset.py:
```diff
@@ -0,0 +1,30 @@
+from token_auth.utils import Util
+
+
+class RequestPasswordResetEmailView(generics.GenericAPIView):
```
Can you please add a doc string (in code documentation) explaining what this test is testing. E.g.: Given user that is registered, when _______, then ______. Just an example, you don't need to follow this format.

DhruviJ17 replied:
Done

isabelcosta commented on token_auth/views/password_reset.py:
```diff
@@ -0,0 +1,30 @@
+            redirect_url = request.data.get('redirect_url', '')
+            absurl = 'http://'+current_site + relativeLink
+            email_body = 'Hello, \n Use link below to reset your password  \n' + \
+                absurl+"?redirect_url="+redirect_url
```
Can you use double quotes in a consistent manner, please? By this I mean, if you use double-quote here, then use it everywhere on this file.

DhruviJ17 replied:
Done

isabelcosta commented on token_auth/serializers/login.py:
```diff
@@ -0,0 +1,46 @@
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+
+
+class LoginSerializer(serializers.ModelSerializer):
```
Can you please add doc strings here, giving a brief explanation of what this class is for?

DhruviJ17 replied:
Done

isabelcosta reviewed (changes requested):
Great work @DhruviJ17 ! I've request some minor changes, please let me know when you've addressed them and if you have any questions.

Thank you for your contribution!

isabelcosta commented:
Can you please add the issue number on the PR description? So its reference in the description @DhruviJ17

codesankalp commented:
@DhruviJ17 Please link your pull request with the issue by changing the commit message or pull-request description.
Also, squash your commits :+1: .

isabelcosta commented:
@DhruviJ17 are you still interested in improving this PR? If so could you please edit the PR description so it describes the changes you added, how you tested this and the issue number that you are fixing.
Can you also edit the issue description to follow the issue templates?

If you are not interested anymore it's completely fine, I can close and make the issue available again :)

isabelcosta reviewed (changes requested):
Can you remove the `.DS_Store` files and add a test for this change please?

DhruviJ17 commented:
done👍🏻

codesankalp commented:
@DhruviJ17  `.DS_Store` file is still present. Can you please remove it.

codesankalp commented:
@DhruviJ17 instead of `**/.DS_Store` add this in `.gitignore`
```
.DS_Store
.AppleDouble
.LSOverride
```
I have added extra git ignores so that in the future there will be no problem regarding this.

codesankalp commented:
@DhruviJ17 You have added `.DS_Store`  in `.gitignore` but it is being tracked by git before adding this in `.gitignore` so try to unstage it.
You can use this:
```
git rm --cached -- .DS_Store
git rm --cached -- .token_auth/serializers/.DS_Store
```

Also, I saw that `password_reset.py` doesn't have an extra line at the end. Please add that by following PEP8 guidelines.

devkapilbansal commented on .gitignore:
```diff
@@ -130,3 +130,8 @@ dmypy.json
+
+.DS_Store
+.AppleDouble
+.LSOverride
```
There should be a blank line at the end

devkapilbansal commented on token_auth/serializers/password_reset.py:
```diff
@@ -0,0 +1,12 @@
+    redirect_url = serializers.CharField(max_length=500, required=False)
+    class Meta:
+        fields = ['email']
+
+
```
Only one line is required here

devkapilbansal commented on token_auth/views/login.py:
```diff
@@ -0,0 +1,17 @@
+Function for user login after registration
+"""
+
+class LoginView(generics.GenericAPIView):
+    serializer_class = LoginSerializer
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```
Add a blank line here too

devkapilbansal replied:
I don't think that you need generic view here
```suggestion
class LoginView(APIView):
    def post(self, request):
        serializer = LoginSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        return Response(serializer.data, status=status.HTTP_200_OK)

```

devkapilbansal reviewed (changes requested):
@DhruviJ17 please use the `commit message guidelines` that we have. The commit message 
 should look like:-
```
type: Short Description

Long Description
```

isabelcosta commented on token_auth/views/password_reset.py:
```diff
@@ -0,0 +1,35 @@
+            relativeLink = reverse('password-reset-confirm', kwargs={'uidb64': uidb64, 'token': token})
+            redirect_url = request.data.get('redirect_url', '')
+            absurl = 'http://'+current_site + relativeLink
+            email_body = 'Hello, \n Use link below to reset your password  \n' + \
```
is there a way to put this email template separately?

codesankalp commented:
@DhruviJ17 Can you fix the merge conflict (solution: rebase your branch) and do the required changes as mentioned by @isabelcosta and @devkapilbansal .

DhruviJ17 commented:
> @DhruviJ17 Can you fix the merge conflict (solution: rebase your branch) and do the required changes as mentioned by @isabelcosta and @devkapilbansal .

When i rebased it showed me already up to date. So i just changed the generic view to api view as mentioned above and added the blank lines.

codesankalp commented:
@DhruviJ17 how do you rebase because I am seeing your branch is 11 commits behind?

Also, the requested changes are to add a single empty line at the end of the file and removing other empty lines.

devkapilbansal commented:
> > @DhruviJ17 Can you fix the merge conflict (solution: rebase your branch) and do the required changes as mentioned by @isabelcosta and @devkapilbansal .
> 
> When i rebased it showed me already up to date. So i just changed the generic view to api view as mentioned above and added the blank lines.

@DhruviJ17 before rebasing, update your develop branch first. Running `git fetch upstream develop` will do the needful.

isabelcosta commented:
@DhruviJ17 could you please edit the description of this PR to describe what you have implemented here? It mentions that you added contribution guidelines link
